tfs/src/main/java/hudson/plugins/tfs/model/AbstractHookEvent.java

```diff
@@ -115,52 +128,65 @@ static JSONObject fromResponseContributors(final List<GitStatus.ResponseContribu
-                            final URIish uri = gitCodePushedEventArgs.getRepoURIish();
-
-                            for (final UserRemoteConfig remoteConfig : gitJobSCM.getUserRemoteConfigs()) {
-                                final String jobRepoUrl = remoteConfig.getUrl();
-                                
-                                if (StringUtils.equals(jobRepoUrl, uri.toString())) {
-                                    shouldRun = true;
+                if (repoMatch(gitCodePushedEventArgs, job)) {
+                    TeamPushTrigger pushTrigger = null;
+                    if (gitCodePushedEventArgs instanceof PullRequestMergeCommitCreatedEventArgs) {
+                        pushTrigger = TeamEventsEndpoint.findTrigger(job, TeamPRPushTrigger.class);
+                    } else { // Check whether current job has an EXACT TeamPushTrigger instead of a TeamPRPushTrigger whose type is also TeamPushTrigger.
+                        final List<TeamPushTrigger> listTriggers = TeamEventsEndpoint.findTriggers(job, TeamPushTrigger.class);
+                        if (!listTriggers.isEmpty()) {
+                            for (TeamPushTrigger trigger : listTriggers) {
+                                if (!(trigger instanceof TeamPRPushTrigger)) {
+                                    pushTrigger = trigger;
                                     break;
                                 }
-                            }                            
+                            }
                         }
                     }
-                }
-                
-                TeamPushTrigger pushTrigger = TeamEventsEndpoint.findTrigger(job, TeamPushTrigger.class);               
-                if (shouldRun && pushTrigger == null)
-                    pushTrigger = new TeamPushTrigger(job);                            
-
-                if (pushTrigger != null) {
-                    pushTrigger.execute(gitCodePushedEventArgs, actions, bypassPolling);
-                    if (bypassPolling) {
-                        return new TeamEventsEndpoint.ScheduledResponseContributor(project);
-                    }
-                    else {
-                        return  new TeamEventsEndpoint.PollingScheduledResponseContributor(project);
+                    if (pushTrigger != null) {
+                        pushTrigger.execute(gitCodePushedEventArgs, actions, bypassPolling);
+                        if (bypassPolling) {
+                            return new TeamEventsEndpoint.ScheduledResponseContributor(project);
+                        } else {
+                            return  new TeamEventsEndpoint.PollingScheduledResponseContributor(project);
+                        }
                     }
                 }
             }
         }
+
         return null;
-    }    
-    
+    }
+
+    private Boolean repoMatch(final GitCodePushedEventArgs gitCodePushedEventArgs, final Job job) {
+        if (job instanceof WorkflowJob) {

```

Comment:
I assume this work for non workflow jobs?

Reply:
Good catch. Non-workflow jobs do repo matching check first can only calls triggerJob() if it matches (in pullOrQueueFromEvent()), but we do need a var in triggerJob() to know the status of "this is a non-workflow job whose repo already matches).

Added a new param "final Boolean repoMatch" to triggerJob() to signal that status.